Apps/FlightSoftware/Watchdog/src/main.c

```diff
@@ -20,11 +20,14 @@ extern uint8_t heating;
 uint8_t lastHeater = 0;
 uint8_t heatingControlEnabled = 1;
 
+I2C_Sensors__Readings i2cReadings;
+I2C_Sensors__Status i2c_status = I2C_SENSORS__STATUS__ERROR__NULL;
+
 // @suppress("Invalid arguments")
 
 /* function definitions in ground_cmd.c */
 void parse_ground_cmd(struct buffer *pp);
-void send_earth_heartbeat();
+void send_earth_heartbeat(I2C_Sensors__Readings i2cReadings);

```

Comment:
I will break these out into a separate header in a bit.

Reply:
I like that idea better than what I did. 

I have i2cReadings init as all 0s so in theory this should work too (unless we switch between KEEPALIVE and MISSION, in which case it'll just keep sending the last reading it took.

Reply:
This needs to be passed as a pointer, otherwise we're copying the entire contents of this structure every time we call this function

Reply:
Now passing a pointer, @ohnx if you want to break it into two headers feel free. Right now it will just send zeros in the place of the battery data (unless we switch from MISSION back to KEEPALIVE, in which case it sends the last readings it took)